transaction-view/src/address_table_lookup_meta.rs

```diff
@@ -103,6 +104,55 @@ impl AddressTableLookupMeta {
     }
 }
 
+pub struct AddressTableLookupIterator<'a> {
+    pub(crate) bytes: &'a [u8],
+    pub(crate) offset: usize,
+    pub(crate) num_address_table_lookup: u8,
+    pub(crate) index: u8,
+}
+
+impl<'a> Iterator for AddressTableLookupIterator<'a> {
+    type Item = SVMMessageAddressTableLookup<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.index < self.num_address_table_lookup {
```

Comment:
Can we use zero-copy serde deserialization here instead of a manual implementation? https://serde.rs/lifetimes.html

Reply:
Also looking forward a bit, serde/bincode has us tied to inline serialization of vecs. And while that is how we do it in the current serialization format, I'm not sure that's the ideal way forward - I have some ideas in this area that I'll post sometime soon.

Reply:
Actually, nevermind. It **wont** work here because serde zero-copy stuff only works for byte arrays or strings.
I think I tried this when I first started looking at it, and since `Pubkey` is not a byte array...it's a **wrapper** for a byte array serde isn't able to tell that it is.

Check this simplified example. Having `Deserialize` on `TransactionRef` makes compilation fail
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=34827612cdb1075756d3f26a97f96e2b

Reply:
@jstarry does this make sense? Waiting to merge on resolution of this conversation.

Reply:
I looked into this for awhile and tried creating a custom deserialize impl for a new `PubkeyRef` struct but ran into issues with using bincode to borrow deserialize a fixed length array. Looks like bincode v2 might help in the future with https://docs.rs/bincode/2.0.0-rc.3/bincode/de/trait.Decode.html but for now manual deserialization here is fine with me.

Reply:
You don't need this if here right? You can iterate based on self.offset. In fact
you're not incrementing self.index afaict so that tripped me a bit.